Add support for empty strings

Providing an empty string, like this:
```
foo = ""
# or
foo = ''
```
causes forge to panic with following message:
```
panic: syntax error line <xy> column <xy>: expected ';' or '
```

When I find the time, I'll take a look at this and create a PR.

Cheers

--- a/forge_test.go
+++ b/forge_test.go
@@ -16,6 +16,7 @@ primary {
   string = "primary string value";
   string_with_quote = "some \"quoted\" str\\ing";
   single = 'hello world';
+  empty = '';
   single_with_quote = '\'hello\' "world"';
 
   # Semicolons are optional
@@ -70,6 +71,7 @@ func assertDirectives(values map[string]interface{}, t *testing.T) {
 	assertEqual(primary["string"], "primary string value", t)
 	assertEqual(primary["string_with_quote"], "some \"quoted\" str\\ing", t)
 	assertEqual(primary["single"], "hello world", t)
+	assertEqual(primary["empty"], "", t)
 	assertEqual(primary["single_with_quote"], "'hello' \"world\"", t)
 	assertEqual(primary["integer500"], int64(500), t)
 	assertEqual(primary["float"], float64(80.80), t)

--- a/scanner.go
+++ b/scanner.go
@@ -123,6 +123,10 @@ func (scanner *Scanner) parseNumber(negative bool) {
 func (scanner *Scanner) parseString(delimiter rune) {
 	scanner.curTok.ID = token.STRING
 	scanner.curTok.Literal = ""
+	if scanner.curCh == delimiter {
+		scanner.readRune()
+		return
+	}
 	// Whether or not we are trying to escape a character
 	escape := false
 

--- a/test.cfg
+++ b/test.cfg
@@ -5,6 +5,7 @@ primary {
   string = "primary string value";
   string_with_quote = "some \"quoted\" str\\ing";
   single = 'hello world';
+  empty = '';
   single_with_quote = '\'hello\' "world"';
 
   # Semicolons are optional

